### PR TITLE
add deploy plugin

### PR DESCRIPTION
**Pull request by lowkeyrd**

### CODE REVIEW

File: packages/s-deploy/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  };
+}
+
+export const run = async (inputs: Record<string, any>, context: Record<string, any>, logger: Logger) => {
+  logger.info('start @serverless-cd/s-deploy run');
+  console.log(logger);
+  const props = await getPluginInputs(inputs, context, logger);
+  const deploy = new Deploy(props as IProps, logger);
+  const res = await deploy.run();
+  logger.info('Run @serverless-cd/s-setup end');
```
Comment (AndyManastorm) [inline packages/s-deploy/src/index.ts:27]: s-deploy

File: packages/fc-canary/package-lock.json
Comment (AndyManastorm) [inline packages/fc-canary/package-lock.json:1]: we have no reason to update anything under `./fc-canary` and `./fc-release`

File: packages/s-deploy/src/deploy.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { lodash as _, Logger, fs } from '@serverless-cd/core';
+import execa from 'execa';
+
+export interface IProps {
+  resource?: string; // Specify a resource in s.yaml. Use the command to s xxx deploy.
+  deployFile?: string; // Specific yaml file. Use the command to s deploy -t xxx 
+  deployArgs?: string[]; // Specific custom args. Merge with other command args during deployment.
+  envName?: string; // Specify envionment. Use the command to s deploy --env
+  debugMode?: boolean; // Use the command to s deploy --debug
+  worksapce?: string // worksapce dir.
```
Comment (AndyManastorm) [inline packages/s-deploy/src/deploy.ts:10]: duplicate: IMHO workspace is duplicate with `cwd`.

refer to: https://serverless-cd.cn/docs/dev-guide/toolkit/engine/#stepoptions
image

File: packages/s-deploy/src/deploy.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  private logger: Logger;
+  private props: IProps;
+  constructor(props: IProps, logger: Logger) {
+    this.logger = (logger || console) as Logger;
+    this.props = props
+  }
+
+  async run() {
+    const { resource, deployFile = 's.yaml', deployArgs, envName, debugMode, worksapce } = this.props;
+    if (!fs.existsSync(`${worksapce}/${deployFile}`)) {
```
Comment (AndyManastorm) [inline packages/s-deploy/src/deploy.ts:23]: what if `deployFile` is a relative path. e.g. './resources/s.yaml'.
maybe we should use `path.resolve(workspace, deployFile)` instead.

File: packages/s-deploy/src/deploy.ts (hunk 2 above)
Comment (AndyManastorm) [reply packages/s-deploy/src/deploy.ts:10]: typo: workspace

File: packages/fc-canary/package-lock.json
Comment (lowkeyrd) [reply packages/fc-canary/package-lock.json:1]: image
The current version is 1.0.0. The package.json should be consistent.

File: packages/s-deploy/src/deploy.ts (hunk 2 above)
Comment (lowkeyrd) [reply packages/s-deploy/src/deploy.ts:10]: > duplicate: IMHO workspace is duplicate with `cwd`.
> 
> refer to: https://serverless-cd.cn/docs/dev-guide/toolkit/engine/#stepoptions image

working-director cannot be obtained globally, and engine does not modify process.cwd. Therefore, I need to record this value from the context when constructing input.

**Comment by AndyManastorm** [comment]
LGTM

**Review by AndyManastorm — APPROVED** [review]

**Pull request merged**